Subject: Work around bugs in Sunrex IR Keyboard/Mouse
Date: Sun Mar 13 22:28:09 CET 2005
From: Vojtech Pavlik <vojtech@suse.cz>
References: #72331
===================================================================

ChangeSet@1.2072, 2005-03-13 19:29:10+01:00, vojtech@suse.cz
  input: Workaround for Sunrex K8561 IR Keyboard/Mouse. The mouse
         sends an incorrect ID and wasn't recognized.
  
  Reported-by: Stefan Seyfried <seife@suse.de>
  Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>

 psmouse-base.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletion(-)

diff -Nru a/drivers/input/mouse/psmouse-base.c b/drivers/input/mouse/psmouse-base.c
--- a/drivers/input/mouse/psmouse-base.c	2005-03-13 22:28:13 +01:00
+++ b/drivers/input/mouse/psmouse-base.c	2005-03-13 22:28:13 +01:00
@@ -518,13 +518,16 @@
 /*
  * First, we check if it's a mouse. It should send 0x00 or 0x03
  * in case of an IntelliMouse in 4-byte mode or 0x04 for IM Explorer.
+ * Sunrex K8561 IR Keyboard/Mouse reports 0xff on second and subsequent
+ * ID queries, probably due to a firmware bug.
  */
 
 	param[0] = 0xa5;
 	if (ps2_command(ps2dev, param, PSMOUSE_CMD_GETID))
 		return -1;
 
-	if (param[0] != 0x00 && param[0] != 0x03 && param[0] != 0x04)
+	if (param[0] != 0x00 && param[0] != 0x03 &&
+	    param[0] != 0x04 && param[0] != 0xff)
 		return -1;
 
 /*
